### PR TITLE
TCM-276 | Build dummy Filters screen

### PR DESCRIPTION
**Pull request by devender197**
### Summary

- [x] Dummy videos filter screen and view model is created
- [x] All related models and classes are created.

### Issue

[TCM-276](https://hyfnla.atlassian.net/browse/TCM-276)

### Video/Screenshot

https://user-images.githubusercontent.com/24737954/118567186-30319f80-b72a-11eb-8ecd-a42d14d8f86d.mov

#ScreenShot

Screenshot_1621292882

### CODE REVIEW

File: app/src/main/res/layout/fragment_filters.xml [hunk 1]
```diff
@@ -0,0 +1,78 @@
+<?xml version="1.0" encoding="utf-8"?>
+<com.trx.consumer.common.CommonView xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (VivekV95) [inline app/src/main/res/layout/fragment_filters.xml:2]: This file should be `fragment_filter` right? `fragment_filters` would be for `FiltersFragment`

Comment (VivekV95) [reply app/src/main/res/layout/fragment_filters.xml:2]: Unless we're re-using it

Comment (devender197) [reply app/src/main/res/layout/fragment_filters.xml]: [Done](https://github.com/hyfn/trx-consumer-android/pull/37/commits/a48e0566ed7f6f57e5a4d6d59e9e7403b1529dfd)

File: app/src/main/res/layout/row_video_filter.xml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        android:textSize="14sp"
+        android:layout_marginStart="25dp"
+        android:layout_marginTop="15dp"
+        android:layout_marginBottom="15dp"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent"
+        tools:text="Duration" />
+
+    <com.trx.consumer.common.CommonImageButton
```
Comment (VivekV95) [inline app/src/main/res/layout/row_video_filter.xml:29]: According to [design](https://www.figma.com/file/wMfhxL9w3KOw6okQJtt1OD/All-Screens-(v1.0.0)?node-id=2636%3A2), looks like we need an additional label to display the selected filter

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -156,4 +157,9 @@
     <string name="onboarding_demand_list_text_two">Huge library of exclusive on-demand video workouts from top TRX pros</string>
     <string name="onboarding_demand_list_text_three">Get your heart rate up with tons of diverse workouts to choose from</string>
 
+    <string name="filter_btn_close_label">CLOSE</string>
+    <string name="filter_btn_clear_filters_label">CLEAR FILTERS</string>
+    <string name="filter_btn_apply_label">APPLY</string>
+    <string name="filter_empty_list_label">There are no filter</string>
```
Comment (VivekV95) [inline app/src/main/res/values/strings.xml:163]: "filter" -> "filters"

File: app/src/main/java/com/trx/consumer/screens/filter/VideoFilterAdapter.kt [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    }
+
+    fun update(newItems: List<VideoFilterModel>) {
+        this.items.clear()
+        this.items.addAll(newItems)
+        this.notifyDataSetChanged()
+    }
+
+    fun reset() {
+       //TO-DO
```
Comment (VivekV95) [inline app/src/main/java/com/trx/consumer/screens/filter/VideoFilterAdapter.kt:60]: Let's mark as `//TODO:` so that it's detected by AS

**Review by VivekV95 — CHANGES_REQUESTED** [review]
Couple comments, looking good otherwise

File: app/src/main/res/values/strings.xml (hunk 3 above)
Comment (devender197) [reply app/src/main/res/values/strings.xml]: [Done](https://github.com/hyfn/trx-consumer-android/pull/37/files#diff-5e01f7d37a66e4ca03deefc205d8e7008661cdd0284a05aaba1858e6b7bf9103R165)

File: app/src/main/res/layout/row_video_filter.xml (hunk 2 above)
Comment (devender197) [reply app/src/main/res/layout/row_video_filter.xml:29]: Actually, we have multiple options for filter and user can select all options. So once we will start working on option screen. We can discuss, what to show selected filter label

File: app/src/main/java/com/trx/consumer/screens/filter/VideoFilterAdapter.kt (hunk 4 above)
Comment (devender197) [reply app/src/main/java/com/trx/consumer/screens/filter/VideoFilterAdapter.kt]: [Done](https://github.com/hyfn/trx-consumer-android/pull/37/commits/5cd234ecd2fe8a67f104c9c01592cf1e5028bb92)

File: app/src/main/res/values/strings.xml (hunk 3 above)
Comment (VivekV95) [reply app/src/main/res/values/strings.xml]: I meant the string for `filter_empty_list_label` should be "There are no filters", sorry should've been more specific

Comment (VivekV95) [reply app/src/main/res/values/strings.xml]: `filter` for the string names makes sense

**Review by VivekV95 — CHANGES_REQUESTED** [review]
Just the "filters" comment above + conflicts and should be good to go

File: app/src/main/res/values/strings.xml (hunk 3 above)
Comment (devender197) [reply app/src/main/res/values/strings.xml]: [Done](https://github.com/hyfn/trx-consumer-android/pull/37/files#diff-5e01f7d37a66e4ca03deefc205d8e7008661cdd0284a05aaba1858e6b7bf9103R171)

**Comment by devender197** [comment]
@onodude @VivekV95 for the review. Please let me know any further changes required. @onodude I have inserted the label for selected filter and will handle this on filter options screen. Thank you.

**Review by VivekV95 — APPROVED** [review]

**Review by onodude — APPROVED** [review]

**Pull request merged**